components/predict-form/Predict.tsx

```diff
@@ -423,26 +309,39 @@ export function Predict({
 
   const [showSuggestions, setShowSuggestions] = useState(false)
 
-  function smartUpdateResolveBy(newQuestionValue: string) {
-    if (!touchedFields.resolveBy && smartSetDates) {
-      const dateResult = chrono.parse(newQuestionValue, new Date(), {
-        forwardDate: true,
-      })
-      const newResolveBy =
-        dateResult.length === 1 && dateResult[0].date()
-          ? getDateYYYYMMDD(dateResult[0].date())
-          : undefined
-
-      if (
-        newResolveBy &&
-        newResolveBy !== getDateYYYYMMDD(utcDateStrToLocalDate(resolveByUTCStr))
-      ) {
-        setValue("resolveBy", newResolveBy)
-        setHighlightResolveBy(true)
-        setTimeout(() => setHighlightResolveBy(false), 800)
+  const smartUpdateResolveBy = useCallback(
+    (newQuestionValue: string) => {
+      if (!touchedFields.resolveBy && smartSetDates) {
+        const dateResult = chrono.parse(newQuestionValue, new Date(), {
+          forwardDate: true,
+        })
+        const newResolveBy =
+          dateResult.length === 1 && dateResult[0].date()
+            ? getDateYYYYMMDD(dateResult[0].date())
+            : undefined
+
+        if (
+          newResolveBy &&
+          newResolveBy !==
+            getDateYYYYMMDD(utcDateStrToLocalDate(resolveByUTCStr))
+        ) {
+          setValue("resolveBy", newResolveBy)
+          setHighlightResolveBy(true)
+          setTimeout(() => setHighlightResolveBy(false), 800)
+        }
       }
-    }
-  }
+    },
+    [touchedFields.resolveBy, smartSetDates, resolveByUTCStr, setValue],
+  )
+
+  const questionRef = useRef(question)
+  useEffect(() => {
+    if (questionRef.current === question) return
+
+    questionRef.current = question
+    smartUpdateResolveBy(question)
```

Comment:
Previously this was called manually after changes that affect it, I moved it to a `useEffect` so that external callers of `usePredictForm` don't need to handle this